#VRML_SIM R2019a utf8
WorldInfo {
  info [
    "Example use of a LightSensor device."
  ]
  title "Light Sensor"
  basicTimeStep 100
}
Viewpoint {
  orientation -0.9887908391693274 -0.09129990215913311 0.11813976570380484 1.5236374708490001
  position -3.026483919910543 13.153421932437013 4.174548691451628
}
TexturedBackground {
}
RectangleArena {
  translation 1.1499999999999997 0 0
  floorSize 10 10
}
DEF LAMP Solid {
  translation -0.9136010555297571 1.4867519920334251 -1.815174979700374
  rotation 0 0 -1 0
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        roughness 1
        metalness 0
        emissiveColor 1 1 1
      }
      geometry Sphere {
        radius 0.01
      }
    }
    PointLight {
      attenuation 0 0 12.56
      intensity 5
    }
  ]
  name "light"
  translationStep 0
  rotationStep 0
}
DEF ROBOT Robot {
  translation -1.7200000000000009 -0.000761012 4
  rotation 0 -1 0 1.570796360359173
  children [
    DistanceSensor {
      translation -0.1414 0.08 0.1414
      rotation 0 1 0 -2.3562
      name "SW"
      lookupTable [
        0 0 0
        1 1 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0.1414 0.08 0.1414
      rotation 0 -0.9999999999999999 0 0.7854
      name "SE"
      lookupTable [
        0 0 0
        1 1 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0.1414 0.08 -0.1414
      rotation 0 -0.9999999999999999 0 -0.7854
      name "NE"
      lookupTable [
        0 0 0
        1 1 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation -0.1414 0.08 -0.1414
      rotation 0 1 0 2.3562
      name "NW"
      lookupTable [
        0 0 0
        1 1 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0 0.08 -0.2
      rotation 0 1 0 1.5707962836408276
      name "N"
      lookupTable [
        0 0 0
        1 1 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0 0.08 0.2
      rotation 0 -1 0 1.5707963603591728
      name "S"
      lookupTable [
        0 0 0
        1 1 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation -0.2 0.08 0
      rotation 0 1 0 3.141592644
      name "W"
      lookupTable [
        0 0 0
        1 1 0
      ]
      type "laser"
      aperture 1.57
    }
    DistanceSensor {
      translation 0.2 0.08 0
      name "E"
      lookupTable [
        0 0 0
        1 1 0
      ]
      type "laser"
    }
    Transform {
      translation 0 0.0415 0
      scale 2 1 2
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.1
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 6333.832976912349
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.04499999435082294 0.025032098706932546 -1.3221645784216788e-05
        rotation -0.9999999999994527 4.5476043356201e-07 -9.423551615213441e-07 1.6807025825514796
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 6527.248410448452
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.04499999436729418 0.02501954978998345 2.8685713649033674e-05
        rotation 0.9999999999975178 2.1638173922392024e-06 -5.315118930197683e-07 5.5684737111827936
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    Transform {
      translation 0.02 0.063 -0.0385
      rotation 0 1 0 2.7489
      children [
        DEF UNUSED_EYE Group {
          children [
            Transform {
              translation 0 0 -1e-05
              rotation 0 0 1 0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
            Transform {
              rotation 0 0 1 -0.8
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.975691 0.981481 0.0252992
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.02 0.005 0.005
                  }
                }
              ]
            }
          ]
        }
      ]
    }
    Transform {
      translation -0.02 0.063 -0.0385
      rotation 0 1 0 0.3927
      children [
        USE UNUSED_EYE
      ]
    }
    LightSensor {
      translation 0.2 0.08 0
      rotation 0 0 1 1.5707959999999999
      name "lsR"
    }
    LightSensor {
      translation 0 0.08 -0.2
      rotation 0 0 1 1.5707959999999999
      name "lsF"
    }
    LightSensor {
      translation 0 0.08 0.2
      rotation 0 0 1 1.5707959999999999
      name "lsB"
    }
    LightSensor {
      translation 0 0.08 0
      rotation 0 0 1 1.5707959999999999
      name "lsC"
    }
    LightSensor {
      translation 0.01 0.08 0
      rotation 0 0 1 1.5707959999999999
      name "Ori0"
    }
    LightSensor {
      translation 0.00866 0.08 0.005
      rotation 0 0 1 1.5707959999999999
      name "Ori1"
    }
    LightSensor {
      translation 0.005 0.08 0.00866
      rotation 0 0 1 1.5707959999999999
      name "Ori2"
    }
    LightSensor {
      translation 0 0.08 0.01
      rotation 0 0 1 1.5707959999999999
      name "Ori3"
    }
    LightSensor {
      translation -0.005 0.08 0.00866
      rotation 0 0 1 1.5707959999999999
      name "Ori4"
    }
    LightSensor {
      translation -0.00866 0.08 0.005
      rotation 0 0 1 1.5707959999999999
      name "Ori5"
    }
    LightSensor {
      translation -0.01 0.08 0
      rotation 0 0 1 1.5707959999999999
      name "Ori6"
    }
    LightSensor {
      translation -0.00866 0.08 -0.005
      rotation 0 0 1 1.5707959999999999
      name "Ori7"
    }
    LightSensor {
      translation -0.005 0.08 -0.00866
      rotation 0 0 1 1.5707959999999999
      name "Ori8"
    }
    LightSensor {
      translation 0 0.08 -0.01
      rotation 0 0 1 1.5707959999999999
      name "Ori9"
    }
    LightSensor {
      translation 0.005 0.08 -0.00866
      rotation 0 0 1 1.5707959999999999
      name "Ori10"
    }
    LightSensor {
      translation 0.00866 0.08 -0.005
      rotation 0 0 1 1.5707959999999999
      name "Ori11"
    }
    LightSensor {
      translation -0.2 0.08 0
      rotation 0.6215150049313568 0.6215148018234582 -0.47690507415996647 4.031596023374673
      name "lsL"
    }
    DEF SMILE Transform {
      translation 0 0.025 -0.036
      rotation 0.7745972408364944 0.44721228048294853 0.4472139204829502 1.8234794
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
  ]
  name "MyBot"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "light_sensor"
  supervisor TRUE
}
